lib/train/platforms/detect/specifications/os.rb

```diff
@@ -27,12 +27,27 @@ def self.load
       # unix master family
       plat.family('unix')
           .detect {
-            if unix_uname_s =~ /./
+            # we want to catch a special case here where cisco commands
+            # don't return an exit status and still print to stdout
+            if unix_uname_s =~ /./ && !unix_uname_s.start_with?('Line has invalid autocommand ')
               @platform[:arch] = unix_uname_m
               true
             end
           }
 
+      # cisco_ios family
+      plat.family('cisco').title('Cisco Family')
+          .detect {
+            true
```

Comment:
I wonder if we shouldn't change this to be something like `!cisco_show_version.nil?` so we don't attempt to do Cisco detection when not necessary.

Reply:
let me do that in the follow-up too